src/Orleans/Logging/LogWriters.cs

```diff
@@ -159,26 +159,29 @@ public class LogWriterToTrace : LogWriterBase, IFlushableLogConsumer
         /// <summary>Write the log message for this log.</summary>
         protected override void WriteLogMessage(string msg, Logger.Severity severity)
         {
+            var sev = SeverityLevel.Information;
+
             switch (severity)
             {
                 case Logger.Severity.Off:
-                    break;
+                    return;
                 case Logger.Severity.Error:
-                    Trace.TraceError(msg);
+                    sev = SeverityLevel.Error;
                     break;
                 case Logger.Severity.Warning:
-                    Trace.TraceWarning(msg);
+                    sev = SeverityLevel.Warning;
                     break;
                 case Logger.Severity.Info:
-                    Trace.TraceInformation(msg);
+                    sev = SeverityLevel.Information;
                     break;
                 case Logger.Severity.Verbose:
                 case Logger.Severity.Verbose2:
                 case Logger.Severity.Verbose3:
-                    Trace.WriteLine(msg);
+                    sev = SeverityLevel.Verbose;
                     break;
             }
-            Flush();
+            //Forward log calls to Telemetry API.
+            Logger.TelemetryConsumers.OfType<TraceTelemetryConsumer>().AsParallel().ForAll(tc => tc.TrackTrace(msg, sev));
```

Comment:
True. `AsParallel` uses default scheduler, which is thread pool, and not current scheduler. It may not be important in this particular case. But then why even bother doing it in parallel? Do you expect it to be a heavy operation? I would say if it is, then the TelemetryConsumer itself should switch threads/buffer/ do what ever it needs to, but return quickly.